src/structs.jl

```diff
@@ -110,11 +110,11 @@ consumption/production rates (accessible in field `stoichiometry`), lower/upper
 bounds (in tuples `lb` and `ub`, with unit names), and objective coefficient
 (`oc`). Also may contains `notes` and `annotation`.
 """
-struct Reaction
-    stoichiometry::Dict{String,Float64}
-    lb::Tuple{Float64,String}
-    ub::Tuple{Float64,String}
-    oc::Float64
+mutable struct Reaction
+    stoichiometry::Dict{String,Float64}  # PL: Why do we allow/require non-integer stoichiometries?
```

Comment:
Unfortunately it seems nature is not made of integers, at least not at this zoom level. :D

SBML allows full floats here and many models use that. (There's even functionality that uses additional denominators to write `0.33333333` precisely as `1/3`, but let's pretend that one doesn't exist.)

Reply:
In that case I think SBML allows stuff that nature doesn't (what on earth is a half-molecule?). But they sure have their reasons (e.g. some people might want to 1*H + 0.5*O -> 0.5*H2O). Thanks for the explanation, I was just puzzled.